src/main/java/com/google/devtools/build/lib/sandbox/SandboxModule.java

```diff
@@ -238,8 +238,31 @@ private void setup(CommandEnvironment cmdEnv, SpawnStrategyRegistry.Builder buil
     // previous builds. However, on the very first build of an instance of the server, we must
     // wipe old contents to avoid reusing stale directories.
     if (firstBuild && sandboxBase.exists()) {
-      cmdEnv.getReporter().handle(Event.info("Deleting stale sandbox base " + sandboxBase));
-      sandboxBase.deleteTree();
+      if (options.asyncFirstBuildDelete && (treeDeleter instanceof AsynchronousTreeDeleter)) {
```

Comment:
Do we need to check for the `treeDeleter` to be a `AsynchronousTreeDeleter`? Seems like the logic below works regardless.